openshift-client/src/main/java/io/fabric8/openshift/client/internal/OpenShiftOAuthInterceptor.java

```diff
@@ -122,5 +109,37 @@ private  String authorize() {
             throw KubernetesClientException.launderThrowable(e);
         }
     }
+
+  private static Response tryWithToken(Chain chain, Request request, Response response, String token) throws IOException {
+    Request.Builder builder = request.newBuilder();
+
+    //If token was obtain, then retry request using the obtained token.
+    if (Utils.isNotNullOrEmpty(token)) {
+      // Close the previous response to prevent leaked connections.
+      response.body().close();
+      setAuthHeader(builder, token);
+      request = builder.build();
+      return chain.proceed(request); //repeat request with new token
+    } else {
+      return response;
+    }
+  }
+
+  private void updateTokenOnSuccess(Response response, String token) throws IOException {
+    if (!isUnauthorized(response)) {
+      oauthToken.set(token);
+    }
+  }
+
+  private static void setAuthHeader(Request.Builder builder, String token) {
+    if (token != null) {
+      builder.header(AUTHORIZATION, String.format("Bearer %s", token));
+    }
+  }
+
+  private static boolean isUnauthorized(Response response) {
+    return response.code() == 401 || response.code() == 403;
```

Comment:
Please use `HttpURLConnection.HTTP_OK` and `HttpURLConnection.HTTP_FORBIDDEN` just to be a bit clearer what these mean.